web/src/main/java/org/akaza/openclinica/control/submit/ListStudySubjectTableFactory.java

```diff
@@ -94,9 +94,34 @@
     // To avoid showing title in other pages, the request element is used to determine where the request came from.
     public TableFacade createTable(HttpServletRequest request, HttpServletResponse response) {
         locale = LocaleResolver.getLocale(request);
-        session = request.getSession();
-        TableFacade tableFacade = getTableFacadeImpl(request, response);
-        tableFacade.setStateAttr("restore");
+        session = request.getSession();      
+     
+        // new setting
+        TableFacade tableFacade = null;
+        String matchedStudyOID = null;
+        if(request.getParameter("maxRows") != null) {        	
+        	tableFacade = prepareNewTableFacade(request, response);
+        }else {
+        	// restore from the session   	        	 
+        	 TableFacade tableFacadeInSession = (TableFacade) request.getSession().getAttribute("LSStableFacade");
+        	 tableFacade = getTableFacadeImpl(request, response);
+        	 matchedStudyOID = (String) request.getSession().getAttribute("matchedStudyOID");
+        	 String currentStudyOID = getStudyBean().getOid();
+        	 
+        	 if(matchedStudyOID != null && matchedStudyOID.equals(currentStudyOID)) {
+        		 if(tableFacadeInSession != null) {
+            		 Limit limitInSession = tableFacadeInSession.getLimit();
+        	         
+            		 tableFacade.setStateAttr("restore");
```

Comment:
does this need to have a name more specific to the matrix so it doesn't interfere with other tables?

Reply:
tableFacade.setStateAttr("restore")  --- this is existing code/logic, and "restore" is under specfic tableFacade.